exchange/bitswap/network/ipfs_impl.go

```diff
@@ -174,7 +178,7 @@ func (bsnet *impl) FindProvidersAsync(ctx context.Context, k *cid.Cid, max int)
 
 // Provide provides the key to the network
 func (bsnet *impl) Provide(ctx context.Context, k *cid.Cid) error {
-	return bsnet.routing.Provide(ctx, k, true)
```

Comment:
Does this call here actually send out a message?

Reply:
It probably does, I'll try to write a test for that

Reply:
We probably want to move away from that now. The equivalent of setting the bsnet.routing.Provide final parameter to 'false'

Reply:
Agreed, though it may be worth making this a configurable thing.

It would be best to have the strategies apply here, though it would require a lot of refactoring to make it fit into the code base nicely.